landingai/storage/data_access.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import tempfile
+from pathlib import Path
+from typing import Optional
+from urllib.parse import urlparse
+
+from typing import Optional
+
+
+from landingai.io import read_file
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def download_public_file(
+    url: str,
+    filename: Optional[Path] = None,
+) -> Path:
+    """Download a file from a public url. This function will follow redirects
+
+    Parameters
+    ----------
+    url : str
+        Source url
+    filename : Optional[Path], optional
+        The file name can include a local path. If no filename is provided a temporary one will be created and when possible the extension of the downloaded file will be used
+
+    Returns
+    -------
+    Path
+        Path to the downloaded file
+    """
+    # TODO: It would be nice for this function to not re-download if the src has not been updated
+    data_bytes, server_filename = read_file(url)  # Fetch the file
+    if filename is not None:
+        with open(str(filename), "wb") as f:
+            f.write(data_bytes)
+
+    else:
+        with tempfile.NamedTemporaryFile(
+            suffix=f"--{server_filename}", delete=False
+        ) as f:
+            f.write(data_bytes)
+    return f.name
+
+
+def get_local_uri(uri: str) -> Path:
```

Comment:
how about name it to `convert_local_path()`? or merge the the functionality with `download_file()`?
the function name could be confusing as it could be downloading file. (My first impression of this function is that it just parses a local file path out of a uri.)

Reply:
it's worth a unit test.

Reply:
Added test